### DISCUSSION TITLE
[ideas]

### DISCUSSION THREAD

**Discussion by ginolhac** [ropensci/targets#1020] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hello William,

For `tar_combine()` in the `targets` [manual](https://books.ropensci.org/targets/static.html#combine) the target to act on is specified with indices. Like this:

``` r
combined <- tar_combine(
  combined_summaries,
  mapped[[2]], 
  command = dplyr::bind_rows(!!!.x, .id = "method")
)
```

To me, this is quite hazardous, and I am now using **names**, since any additional targets is shifting the numbering and leads to error. So concretely, I am using this for your example:

``` r
  mapped[["summary"]], 
```

Do you see any side effects? If not, I would recommend to update the manual. When I was implementing dynamic within static. I was reading files and forgot to iterate with `format = file` with a stem, to be able to also read files with dynamic branching. Adding the correct target changed the numbering and with combining on names, it is safe from those changes.

Many thanks again for the wonderful documentation and work you do, truly amazing piece of software.

**Comment by wlandau** [comment]
Thanks for letting me know @ginolhac, and as always, thanks for the kind words. I implemented the change you suggested in https://github.com/ropensci-books/targets/commit/1984d67d9d88a961479e08448f5d57364113e6b7.

**Reply by ginolhac** [reply]
thanks a lot William. sorry for not opening this discussion in the right repo. The usual Github logo in Books is not shown in the manual, and forgot to check in `ropensci-books` (explained why I didn't find it in `ropensci`)
